helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -349,4 +338,61 @@ private IdealState generateIdealStateWithAssignment(String resourceName,
     }
     return preferenceList;
   }
+
+  private Map<String, ResourceAssignment> getBaseline(
+      AssignmentMetadataStore assignmentMetadataStore, CurrentStateOutput currentStateOutput,
+      Set<String> resources) throws HelixRebalanceException {
+    Map<String, ResourceAssignment> currentBaseline;
+    try {
+      currentBaseline = assignmentMetadataStore.getBaseline();
+    } catch (HelixException hex) {
+      LOG.warn("Failed to get the current baseline assignment. Use the current states instead",
+          hex);
+      currentBaseline = getCurrentStateAssingment(currentStateOutput, resources);
+    } catch (Exception ex) {
+      throw new HelixRebalanceException(
+          "Failed to get the current baseline assignment because of unexpected error.",
+          HelixRebalanceException.Type.INVALID_REBALANCER_STATUS, ex);
+    }
+    return currentBaseline;
+  }
+
+  private Map<String, ResourceAssignment> getBestPossibleAssignment(
+      AssignmentMetadataStore assignmentMetadataStore, CurrentStateOutput currentStateOutput,
+      Set<String> resources) throws HelixRebalanceException {
+    Map<String, ResourceAssignment> currentBestAssignment;
+    try {
+      // TODO fix the deserialize
```

Comment:
@narendly The test fails here. I have to work around for now. Will talk with you tomorrow about details.

Reply:
Isn't it because you never wrote to bestPossibleAssignment? Didn't we agree that you'll handle it in the rebalancer?

Reply:
Data is there. The deserialize fails. I haven't check details yet.
If you check the code carefully, logic for handling no assignment history has been done.

Reply:
I kind of like the original design... At least that would be easier for us to differentiate if data does not exist or there are some read failure. Let me try to modify it back.

But I guess the deserialize is still there. I will run the test again using my desktop tomorrow.

Reply:
That is exactly the reason I had it so that it returned an empty map.

Reply:
The issus has been fixed.